src/logic/UniqueNamedContainer.java

```diff
@@ -48,56 +48,58 @@ public UniqueNamedContainer() {
      *   There is another item with a different owner and the same original name.
      *
-     * @param item item to be added to the end of this container.
+     * @param item Item to be added to the end of this container.
      */
     public void add(T item) {
         T newItem = (T) item.getCopy();
 
         if (!item.isGlobal()) {
             // Unique name naming rules:
-            //   Same owner and different name: Keep it as is
-            //   Same owner and original name: Add owner to name "owner.n.name" where n is a counter value
+            //   Same owner and different name: keep it as is.
+            //   Different owner and name: keep it as is.
+            //   Different or the same owner and the same original name: add owner to name "owner.n.name" where n is a counter value.
             // Motivation: Machine <= Machine || Machine. Here the owner is Machine but their clocks are different.
-            //   Different owner and name: Keep it as is
-            //   Different owner and same original name: Add owner to name "owner.n.name" where n is a counter value
-            List<T> similarOriginalName = items.stream()
+            List<T> sameOriginalName = items.stream()
                     .filter(current -> Objects.equals(current.getOriginalName(), newItem.getOriginalName()))
                     .collect(Collectors.toList());
 
-            if (similarOriginalName.size() != 0) {
+            if (!sameOriginalName.isEmpty()) {
 
-                List<T> similarOwner = similarOriginalName.stream()
+                List<T> sameOwner = sameOriginalName.stream()
                         .filter(current -> Objects.equals(current.getOwnerName(), newItem.getOwnerName()))
                         .collect(Collectors.toList());
 
-                if (similarOwner.size() > 0) {
-                    for (int i = 0; i < similarOwner.size(); i++) {
-                        similarOwner.get(i).setUniqueName(i + 1);
-                    }
-                    newItem.setUniqueName(similarOwner.size() + 1);
+                if (sameOwner.size() == 1) {
+                    sameOwner.get(0).setUniqueName(1);
+                }
+
+                if (!sameOwner.isEmpty()) {
+                    newItem.setUniqueName(sameOwner.size() + 1);
                 } else {
-                    for (T current : similarOriginalName) {
+                    for (T current : sameOriginalName) {
                         current.setUniqueName();
                     }
                     newItem.setUniqueName();
                 }
             }
-        }
 
-        // If the unique name is not present in the set of items then add it
-        Optional<T> existing = findFirstByUniqueName(newItem.getUniqueName());
-        if (existing.isEmpty()) {
             items.add(newItem);
+        } else {
+            // If the unique name of the global item is not present in the set of items then add it.
```

Comment:
By including this in the `else`, the comment became a little bit confusing. The `else` represents those cases where `newItem` is not global. So where is global referring to in the comment?